### ISSUE TITLE
check shape

### ISSUE THREAD

**Issue by MordehayM** [PyTorchLightning/metrics#859]
https://github.com/PyTorchLightning/metrics/blob/21ba6502418f537a7ca3618be0b19f617f83a062/torchmetrics/functional/audio/pit.py#L148
Hi,
I don't understand why the target shape and pred shape must be equal?
This question arises from the fact that the loss can be the categorical cross-entropy with multiple outputs (for each speaker, for instance) and then this constraint does not apply(categorical cross-entropy in Pytorch)

**Comment by SkafteNicki** [comment]
cc: @quancs

**Comment by quancs** [comment]
Hi, you are right this constraint is not designed correctly for all possible use cases... Could you give some example input and output for your use case

**Comment by quancs** [comment]
>  I don't understand why the target shape and pred shape must be equal?
1) the batch dim of the pred and target should be the same by nature
2) the speaker dim should be the same required by PIT
3) For speech separation, the metrics like SDR, SI-SDR, PESQ require the pred and target to have the same shape at the time dim. 
I guess it is (3) could not be applied to other audio sub-domains. Is that right? @MordehayM

**Comment by MordehayM** [comment]
Yes, exactly.
When the loss is categorical cross entropy, the target and pred do not have the same shape, the target's shape is [B, num_speaker, d1..dk] while the pred's shape is [B, num_speaker, C, d1..dk] where C is the number of categories.

**Comment by quancs** [comment]
> Yes, exactly. When the loss is categorical cross entropy, the target and pred do not have the same shape, the target's shape is [B, num_speaker, d1..dk] while the pred's shape is [B, num_speaker, C, d1..dk] where C is the number of categories.

@MordehayM  So, it works for your case if we just check the first two dimensions, batch and speaker?

**Comment by MordehayM** [comment]
Sorry, but I didn't check that. I chose another implementation of PIT.
In my opinion it should work.

**Comment by quancs** [comment]
> In my opinion it should work.
Thanks for your opinions. I will fix this problem. 

> Sorry, but I didn't check that. I chose another implementation of PIT. 
just a little bit curious, could you tell me which one? If we find it's faster, we could implement it in TorchMetrics ^^

**Comment by MordehayM** [comment]
This one:
https://github.com/asteroid-team/pytorch-pit/blob/master/torch_pit/pit_wrapper.py
